core/object/object.cpp

```diff
@@ -527,6 +527,27 @@ void Object::get_property_list(List<PropertyInfo> *p_list, bool p_reversed) cons
 void Object::validate_property(PropertyInfo &p_property) const {
 	_validate_propertyv(p_property);
 
+	if (_extension && _extension->validate_property) {
+		// GDExtension uses a StringName rather than a String for property name.
+		StringName prop_name = p_property.name;
```

Comment:
Regarding these StringName manipulations, should we share similar concerns to other extended Object methods around here? There is this whole preprocessor pragma guard every time we cast to `GDExtensionStringNamePtr`. We don't do it here, not directly at least, but maybe we should? And maybe we should add the same guards?

Reply:
I think the `#pragma` stuff is about the `const`ness in those casts. In the code added in this PR, we don't need to mess with `const`, so I think it's OK? And running locally with GCC, I don't see any warnings.

Reply:
Alright, sounds fine then. I don't really understand this myself, so I'm trusting you :)